auto-send-to/src/main.ts

```diff
@@ -0,0 +1,86 @@
+import { Metadata } from '@devvit/protos';
+import { Devvit, getSetting } from '@devvit/public-api';
+
+Devvit.use(Devvit.Types.HTTP);
+
+const settingOptions: any = [
```

Comment:
Can we just leave the type off and infer it?

Reply:
when I leave the type off I get this error on the `addSettings(settingsOptions)` line
```
Argument of type '({ defaultValue: string; type: string; name: string; label: string; options: { label: string; value: string; }[]; } | { type: string; name: string; label: string; defaultValue?: never; options?: never; })[]' is not assignable to parameter of type 'SettingsFormField[]'.
```

I don't think SettingsFormField[] is exposed as a type I can use yet

Reply:
> SettingsFormField[]

I agree! This should be exposed :disappointed: 

> Is there another way?

My thinking was that we just make this object const like so:

```ts
const settingOptions = [
  {
    defaultValue: 'posts',
    type: 'select',
    name: 'listenerObject',
    label: 'Objects to Send On',
    options: [
      { label: 'Posts', value: 'posts' },
      { label: 'Comments', value: 'comments' },
      { label: 'Posts and Comments', value: 'both' },
    ],
  },
  {
    type: 'string',
    name: 'webhook',
    label: 'Webhook URL',
  },
  {
    type: 'string',
    name: 'flairId',
    label: 'Only send when post has this flair ID (optional, will not send comments too)'
  }
] as const;
```

However, this breaks because the addSettings() input doesn't take a readonly :rage1: I think it's mutable all the way down so spreading the outer array doesn't work either.

Inlining works and it looks like `defaultValue` is wrong? The typing says it's supposed to be an array:

```ts
Devvit.addSettings([
  {
    defaultValue: ['posts'], // <-- I am array?
    type: 'select',
    name: 'listenerObject',
    label: 'Objects to Send On',
    options: [
      { label: 'Posts', value: 'posts' },
      { label: 'Comments', value: 'comments' },
      { label: 'Posts and Comments', value: 'both' },
    ],
  },
  {
    type: 'string',
    name: 'webhook',
    label: 'Webhook URL',
  },
  {
    type: 'string',
    name: 'flairId',
    label: 'Only send when post has this flair ID (optional, will not send comments too)'
  }
]);
```